### PR TITLE
DS-1190-Value type missmatch error when invoking services with Numeric/Blob types

### PR DESCRIPTION
**Pull request by anupama-pathirage**
DS-1190-Value type missmatch error when invoking services with Numeric/Blob types

### CODE REVIEW

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java [hunk 1]
```diff
@@ -751,6 +751,12 @@ private String nextValue() {
                     value = String.valueOf(jsonReader.nextDouble());
                 } else if (valueType.equals("float")) {
                     value = String.valueOf(jsonReader.nextDouble());
+                } else if (valueType.equals("decimal")) {
```
Comment (madhawa-gunasekara) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java]: shall we swap this, for safety of NPE

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java [hunk 2]
```diff
@@ -780,7 +786,13 @@ private boolean validateArgumentTypes(JsonToken tokenType, String expectedType)
             return true;
         } else if (tokenType == JsonToken.NUMBER) {
             if (expectedType.equals("int") || expectedType.equals("long") || expectedType.equals("double")
-                    || expectedType.equals("float")) {
+                    || expectedType.equals("float") || expectedType.equals("decimal") || expectedType.equals("byte")
+                    || expectedType.equals("short")) {
+                return true;
+            }
+        } else if (tokenType == JsonToken.STRING) {
+            if (expectedType.equals("string") || expectedType.equals("date") || expectedType.equals("time")
```
Comment (madhawa-gunasekara) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java]: you can swap these as well
("string").equals(expectedType)

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java (hunk 1 above)
Comment (anupama-pathirage) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java]: Fixed with PR: #95

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java (hunk 2 above)
Comment (anupama-pathirage) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java]: Fixed with PR: #95

**Pull request merged**